packages/syft/src/syft/core/node/common/client_manager/oblv_api.py

```diff
@@ -0,0 +1,155 @@
+        self,
+        deployment: DeploymentClient,
+        dataset: Union[str, TensorWrappedPhiTensorPointer],
+        **kwargs: Any,
+    ) -> Any:
+        content = {
+            "deployment_id": deployment.deployment_id,
+            "client": SyftOblvClient.from_client(deployment.oblv_client),
+        }
+        dataset_id = (
+            dataset if type(dataset) == "str" else dataset.id_at_location.to_string()
+        )
+        content.update({"dataset_id": dataset_id})
+        response = self.perform_api_request(
+            syft_msg=PublishDatasetMessage, content=content
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+        else:
+            dataset_id = getattr(response, "dataset_id")
+            return OblvTensorWrapper(id=dataset_id, deployment_client=deployment)
+
+    def publish_budget(
+        self, deployment_id, publish_request_id, client, **kwargs: Any
+    ) -> Any:
+        response = self.oblv_perform_api_request_without_reply(
+            syft_msg=PublishApprovalMessage,
+            content={
+                "result_id": publish_request_id,
+                "client": SyftOblvClient.from_client(client),
+                "deployment_id": deployment_id,
+            },
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+
+    def publish_request_budget_deduction(
+        self, deployment_id, publish_request_id, budget_to_deduct, client, **kwargs: Any
+    ) -> Any:
+        response = self.oblv_perform_api_request_without_reply(
+            syft_msg=DeductBudgetMessage,
+            content={
+                "result_id": publish_request_id,
+                "client": SyftOblvClient.from_client(client),
+                "budget_to_deduct": budget_to_deduct,
+                "deployment_id": deployment_id,
+            },
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
+
+    def __getitem__(self) -> Any:
+        return self.get()
+
+    def oblv_perform_api_request_without_reply(
+        self,
+        syft_msg: Optional[Type[SyftMessage]],
+        content: Optional[Dict[Any, Any]] = None,
+        timeout: Optional[int] = None,
+    ) -> Any:
+        if syft_msg is None:
+            raise ValueError(
+                "Can't perform this type of api request, the message is None."
+            )
+        else:
+            syft_msg_constructor = syft_msg
+
+        if content is None:
+            content = {}
+        content[RequestAPIFields.ADDRESS] = self.client.address
+
+        signed_msg = syft_msg_constructor(**content).sign(
+            signing_key=self.client.signing_key
+        )  # type: ignore
+        response = self.client.send_immediate_msg_without_reply(
+            msg=signed_msg, timeout=timeout
+        )
+        if isinstance(response, ExceptionMessage):
+            raise response.exception_type
```

Comment:
Q: Can't we simply raise the exception? (ex: reraise response?)

Reply:
We do have a custom __repr__ for exception type, so we tend to raise the exception type.